contracts/SushiMaker.sol

```diff
@@ -1,264 +1,112 @@
-            // eg. USDT - ETH
-            sushiOut = _toSUSHI(
-                weth,
-                _swap(token0, weth, amount0, address(this)).add(amount1)
-            );
-        } else {
-            // eg. MIC - USDT
-            address bridge0 = bridgeFor(token0);
-            address bridge1 = bridgeFor(token1);
-            if (bridge0 == token1) {
-                // eg. MIC - USDT - and bridgeFor(MIC) = USDT
-                sushiOut = _convertStep(
-                    bridge0,
-                    token1,
-                    _swap(token0, bridge0, amount0, address(this)),
-                    amount1
-                );
-            } else if (bridge1 == token0) {
-                // eg. WBTC - DSD - and bridgeFor(DSD) = WBTC
-                sushiOut = _convertStep(
-                    token0,
-                    bridge1,
-                    amount0,
-                    _swap(token1, bridge1, amount1, address(this))
-                );
-            } else {
-                sushiOut = _convertStep(
-                    bridge0,
-                    bridge1, // eg. USDT - DSD - and bridgeFor(DSD) = WBTC
-                    _swap(token0, bridge0, amount0, address(this)),
-                    _swap(token1, bridge1, amount1, address(this))
-                );
-            }
-        }
-    }
-
-    // F1 - F10: OK
-    // C1 - C24: OK
-    // All safeTransfer, swap: X1 - X5: OK
-    function _swap(
-        address fromToken,
-        address toToken,
-        uint256 amountIn,
-        address to
-    ) internal returns (uint256 amountOut) {
-        // Checks
-        // X1 - X5: OK
-        IUniswapV2Pair pair =
-            IUniswapV2Pair(factory.getPair(fromToken, toToken));
-        require(address(pair) != address(0), "SushiMaker: Cannot convert");
-
-        // Interactions
-        // X1 - X5: OK
-        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
-        uint256 amountInWithFee = amountIn.mul(997);
-        if (fromToken == pair.token0()) {
-            amountOut =
-                amountInWithFee.mul(reserve1) /
-                reserve0.mul(1000).add(amountInWithFee);
-            IERC20(fromToken).safeTransfer(address(pair), amountIn);
-            pair.swap(0, amountOut, to, new bytes(0));
-            // TODO: Add maximum slippage?
-        } else {
-            amountOut =
-                amountInWithFee.mul(reserve0) /
-                reserve1.mul(1000).add(amountInWithFee);
-            IERC20(fromToken).safeTransfer(address(pair), amountIn);
-            pair.swap(amountOut, 0, to, new bytes(0));
-            // TODO: Add maximum slippage?
-        }
-    }
-
-    // F1 - F10: OK
-    // C1 - C24: OK
-    function _toSUSHI(address token, uint256 amountIn)
-        internal
-        returns (uint256 amountOut)
-    {
-        // X1 - X5: OK
-        amountOut = _swap(token, sushi, amountIn, bar);
-    }
+    function burnPair(IUniswapV2Pair pair) public {
```

Comment:
This needs to be `onlyOperator` or else someone can unbalance the pool, make us withdraw liquidity, and then rebalance it for a potential profit as they'd have to spend fewer tokens to rebalance it (since they have to rebalance less liquidity).

Reply:
Though we only hold minority position of pools but  can obviously also guard it :)